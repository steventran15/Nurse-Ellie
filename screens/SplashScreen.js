import React from 'react';
import { View, Image, TouchableOpacity, Dimensions, StyleSheet } from 'react-native';
import PatientStyles from '../styles/PatientStyleSheet';
import Background from '../components/background';
import NurseEllieLogo from '../assets/images/nurse-ellie-logo.svg';

const SplashScreen = ({ navigation }) => {
    return (
        <View style={[PatientStyles.background, {justifyContent: 'center'}]}>
            <Background />
            <View style={styles.logoCircle}>
                <NurseEllieLogo style={styles.logo} />
            </View>
            <TouchableOpacity style={styles.button} onPress={() => navigation.navigate('SignInScreen')}>
                <Image style={{ height: screenHeight * 0.09 }} source={require('../assets/android/drawable-mdpi/g-entry-arrow.png')} />
            </TouchableOpacity>
        </View>
    )
}

var screenHeight = Dimensions.get("window").height;
var screenWidth = Dimensions.get("window").width;

const styles = StyleSheet.create({
    logoCircle: {
        height: 300,
        width: 300,
        borderRadius: 300 / 2,
        backgroundColor: 'rgba(255, 255, 255, 1)',
        position: 'absolute',
        left: screenWidth / 2 - 150,
        justifyContent: 'center',
        alignItems: 'center'
    },
    logo: {
        height: 260,
        width: 260,
        left: 15
    },
    button: {
        position: 'absolute',
        right: screenWidth / 8,
        bottom: screenHeight / 4.5
    }
});

export default SplashScreen;