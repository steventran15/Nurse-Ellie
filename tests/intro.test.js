import React from 'react';
import renderer from 'react-test-renderer';

import Background from '../components/background';
import BackgroundHP from '../components/backgroundHP';
import SplashScreen from '../screens/SplashScreen';

test('Patient background component renders correctly', ()=>{
    const wrapper = renderer.create(<Background/>).toJSON();
    expect(wrapper).toMatchSnapshot();
});

test('Health Professional background component renders correctly', ()=>{
    const wrapper = renderer.create(<BackgroundHP/>).toJSON();
    expect(wrapper).toMatchSnapshot();
});

test('Splash Screen renders correctly', ()=>{
    const wrapper = renderer.create(<SplashScreen/>);
});
