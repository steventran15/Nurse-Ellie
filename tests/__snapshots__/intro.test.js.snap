// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Health Professional background component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#4285C8",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(51, 112, 172, 0.8)",
        "borderRadius": 250,
        "height": 500,
        "position": "absolute",
        "right": -375,
        "top": 667,
        "width": 500,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.3)",
        "borderRadius": 30,
        "height": 60,
        "left": 10,
        "position": "absolute",
        "top": 30,
        "width": 60,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.3)",
        "borderRadius": 125,
        "height": 250,
        "left": 150,
        "position": "absolute",
        "top": 222.33333333333334,
        "width": 250,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.3)",
        "borderRadius": 300,
        "height": 600,
        "left": 300,
        "position": "absolute",
        "top": -333.5,
        "width": 600,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.3)",
        "borderRadius": 175,
        "height": 350,
        "left": -250,
        "position": "absolute",
        "top": 667,
        "width": 350,
      }
    }
  />
</View>
`;

exports[`Patient background component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#42C86A",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(62, 185, 99, 0.8)",
        "borderRadius": 250,
        "height": 500,
        "position": "absolute",
        "right": -375,
        "top": 667,
        "width": 500,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.3)",
        "borderRadius": 30,
        "height": 60,
        "left": 10,
        "position": "absolute",
        "top": 30,
        "width": 60,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.3)",
        "borderRadius": 125,
        "height": 250,
        "left": 150,
        "position": "absolute",
        "top": 222.33333333333334,
        "width": 250,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.3)",
        "borderRadius": 300,
        "height": 600,
        "left": 300,
        "position": "absolute",
        "top": -333.5,
        "width": 600,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.3)",
        "borderRadius": 175,
        "height": 350,
        "left": -250,
        "position": "absolute",
        "top": 667,
        "width": 350,
      }
    }
  />
</View>
`;
